docs/tutorials/quesion-answering-bot.md

```diff
@@ -0,0 +1,217 @@
+```
+
+Now that we have seen the overall structure of the approach and have defined our `Flows`, we can code up the `Executor`s.
+
+## Building the Executor to Generate Potential Questions 
+
+The first `Executor`, that we implement, is the `QuestionGenerator`. It is basically a wrapper around the model that predicts potential questions, which a given piece of text can answer. </br>
+Apart from that, all that it does is looping all provided parts of input text. After potential questions are predicted for each of the inputs, they are stored as `chunks` alongside the original text. 
+
+``` python 
+class QuestionGenerator(Executor): 
+
+    @requests
+    def doc2query(self, docs: DocumentArray, **kwargs):
+        """Generates potential questions for each answer"""
+        
+        # Load pretrained doc2query models
+        self._tokenizer = T5Tokenizer.from_pretrained(
+            'castorini/doc2query-t5-base-msmarco')
+        self._model = T5ForConditionalGeneration.from_pretrained(
+            'castorini/doc2query-t5-base-msmarco')
+
+        for d in docs:
+            input_ids = self._tokenizer.encode(
+                d.content, return_tensors='pt')
+            # Generte potential queries for each piece of text
+            outputs = self._model.generate(
+                input_ids=input_ids,
+                max_length=64,
+                do_sample=True,
+                num_return_sequences=10,
+            )
+            # Decode the outputs ot text and store them 
+            for output in outputs:
+                question = self._tokenizer.decode(
+                    output, skip_special_tokens=True).strip()
+                d.chunks.append(Document(text=question))
+```
+
+We try to give credit where credit is due and want to mention the paper, that introduced the doc2query approach [here](https://arxiv.org/pdf/1904.08375.pdf).
+
+## Building the Encoder
+The next step is to build the `Executor` that we will use to create vector representations from human-readable text. 
+
+```python 
+class TextEncoder(Executor):
+    
+    @requests(on=['/search', '/index'])
+    def encode(self, docs: DocumentArray, 
+               traversal_paths: Tuple[str] = ('r',), **kwargs):
+        """Wraps encoder from sentence-transformers package"""   
+        target = docs.traverse_flat(traversal_paths)
+
+        model = SentenceTransformer(
+            'paraphrase-mpnet-base-v2', device="cpu", cache_folder=".")
+    
+        with torch.inference_mode():
+            embeddings = model.encode(target.texts)
+            target.embeddings = embeddings
+```
+Similar to the `QuestionGenerator` the `TextEncoder` is simply a wrapper around the SentenceTransformer from the sentence_transformer package. When provided with a `DocumentArray` containing text, it will encode the text of each element and store the result in the `embedding` attribute it creates. </br>
+Now let's move on to the last part and create the indexer. 
```

Comment:
```suggestion
Similar to the `QuestionGenerator` the `TextEncoder` is simply a wrapper around the SentenceTransformer from the sentence_transformer package. When provided with a `DocumentArray` containing text, it will encode the text of each element and store the result in the `embedding` attribute it creates.

Now let's move on to the last part and create the indexer. 
```